src/sbvr-api/sbvr-utils.ts

```diff
@@ -0,0 +1,2063 @@
+				table.localFields[odataName] = true;
+			}
+		}
+	}
+	return table.localFields;
+};
+const getFetchProcessingFields = (
+	table: AbstractSQLCompiler.AbstractSqlTable,
+) => {
+	if (table.fetchProcessingFields == null) {
+		table.fetchProcessingFields = _(table.fields)
+			.filter(
+				({ dataType }) =>
+					sbvrTypes[dataType] != null &&
+					sbvrTypes[dataType].fetchProcessing != null,
+			)
+			.map(({ fieldName, dataType }) => {
+				const odataName = sqlNameToODataName(fieldName);
+				return [odataName, sbvrTypes[dataType].fetchProcessing];
+			})
+			.fromPairs()
+			.value();
+	}
+	return table.fetchProcessingFields!;
+};
+const processOData = (
+	vocab: string,
+	abstractSqlModel: AbstractSQLCompiler.AbstractSqlModel,
+	resourceName: string,
+	rows: _db.Result['rows'],
+): Promise<number | _db.Row[]> => {
+	if (rows.length === 0) {
+		return Promise.resolve([]);
+	}
+
+	if (rows.length === 1) {
+		if (rows[0].$count != null) {
+			const count = parseInt(rows[0].$count, 10);
+			return Promise.resolve(count);
+		}
+	}
+
+	const sqlResourceName = resolveSynonym({
+		abstractSqlModel,
+		vocabulary: vocab,
+		resourceName,
+	});
+	const table = abstractSqlModel.tables[sqlResourceName];
+
+	const odataIdField = sqlNameToODataName(table.idField);
+	const instances = rows.map(instance => {
+		instance.__metadata = {
+			// TODO: This should support non-number id fields
+			uri: odataResourceURI(vocab, resourceName, +instance[odataIdField]),
+			type: '',
+		};
+		return instance;
+	});
+
+	let instancesPromise = Promise.resolve();
+
+	const localFields = getLocalFields(table);
+	// We check that it's not a local field, rather than that it is a foreign key because of the case where the foreign key is on the other resource
+	// and hence not known to this resource
+	const expandableFields = _.filter(
+		_.keys(instances[0]),
+		fieldName =>
+			!_.startsWith(fieldName, '__') && !localFields.hasOwnProperty(fieldName),
+	);
+	if (expandableFields.length > 0) {
+		instancesPromise = Promise.map(instances, instance =>
+			Promise.map(expandableFields, fieldName =>
+				checkForExpansion(
+					vocab,
+					abstractSqlModel,
+					sqlResourceName,
+					fieldName,
+					instance,
+				),
+			),
+		).return();
+	}
+
+	const fetchProcessingFields = getFetchProcessingFields(table);
+	const processedFields = _.filter(
+		_.keys(instances[0]),
+		fieldName =>
+			!_.startsWith(fieldName, '__') &&
+			fetchProcessingFields.hasOwnProperty(fieldName),
+	);
+	if (processedFields.length > 0) {
+		instancesPromise = instancesPromise
+			.then(() => {
+				return Promise.map(instances, instance => {
+					return Promise.map(processedFields, resourceName => {
+						return fetchProcessingFields[resourceName](
+							instance[resourceName],
+						).then(result => {
+							instance[resourceName] = result;
+						});
+					});
+				});
+			})
+			.return();
+	}
+
+	return instancesPromise.return(instances);
+};
+
+// export const runRule = (() => {
```

Comment:
I forgot I commented this, it needs fixing but is unused in the odata chain (it enables queries via sbvr fragments instead)